src/webui/www/private/scripts/client.js

```diff
@@ -95,7 +95,51 @@ const getShowFiltersSidebar = function() {
     return (show === null) || (show === 'true');
 };
 
+function isValidIpv4Address(ip) {
+    // Regular expression to validate IPv4 address format
+    const ipv4Regex = /^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$/;
+    return ipv4Regex.test(ip);
+}
+
+function isValidIpv6Address(ip) {
+    // Regular expression to validate IPv6 address format
+    const ipv6Regex = /^(?:[A-F0-9]{1,4}:){7}[A-F0-9]{1,4}$/i;
+    return ipv6Regex.test(ip);
+}
+
+// getHost emulate the GUI version `QString getHost(const QString &url)`
+function getHost(url) {
+    // We want the domain + tld. Subdomains should be disregarded
+    // If failed to parse the domain or IP address, original input should be returned
+
+    try {
+        const parsedUrl = new URL(url);
+        // host: "example.com:8443"
+        // hostname: "example.com"
+        const host = parsedUrl.hostname;
+        if (!host) {
+            return url;
+        }
+
+        // host is in IP format
+        if (isValidIpv4Address(host) || isValidIpv6Address(host)) {
+            return host;
+        }
+
+        return host.split(/\./).slice(-2).join('.');
+    }
+    catch (error) {
+        console.error(error);
+        return url;
+    }
+}
+
+
 function genHash(string) {
+    if (string.startsWith('http://') || string.startsWith('https://') || string.startsWith('udp://')) {
+       string = getHost(string);
+    }
```

Comment:
I like `genHash()` to be kept simple so maybe create a helper function for generating hash for host? Such as:
```js
function genHostHash(string) {
    if (string.startsWith('http://') || string.startsWith('https://') || string.startsWith('udp://'))
       return genHash(getHost(string));
    return -1;
}
```
And `genHostHash()` actually looks kind of silly maybe there will be better ideas..

Reply:
Although unlikely, scheme (by definition) seems to be case-insensitive. Better use `const lowerString = string.toLowerCase();` before comparing them.

Reply:
resolve via https://github.com/qbittorrent/qBittorrent/pull/18190/commits/7401709dcc20f56ca69c24f2ea5072f40e912e28

Reply:
> I like `genHash()` to be kept simple

👍 


> And `genHostHash()` actually looks kind of silly maybe there will be better ideas..

This is because you want to combine essentially separate actions in an unnatural way. It would be much clearer to use them in a chain, i.e. `genHash(getHost(string))`.

Reply:
done